### PR TITLE
9-modeling users (sign up and sign in)

### PR DESCRIPTION
**Pull request by godfreym**
Issue #9 

## Note:
Please chill out and have a cup of :coffee: while reviewing this PR, Devise is packed with many files and they are long so at some point the PR might seem big but in reality it is NOT, just devise files which by the way are important and needed to accomplish the mission successfully and in the right way.. Have fun!

## Story:

AS A User of the site
I WANT To be able to sign up or sign in to the site
SO THAT I can use my credentials to sign in to the site at any time I need to

## Acceptance Criteria:
- [x] GIVEN I am the user and want to use the site WHEN I access the site THEN I should be able to sign up to the site and sign in at any time after I have already signed up

## Task Involved:
- [x] Add ` gem 'devise' ` on the Gemfile and `$ binit`
- [x] Installing devise by `$ rails generate devise:install`
- [x] Generating the devise User model by running `$ rails generate devise User`
- [x] Integrating `application_controller` with devise by adding `before_action :authenticate_user!` on it in order to authenticate users before they can access the site 
- [x] Generating devise views by `$ rails generate devise:views` so that I can customize them
- [x] Associating the User model with Company

### CODE REVIEW

**Comment by godfreym** [comment]
@leonardk1 Please Review

**Comment by godfreym** [comment]
@leonardk1 Please Review

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<div class="top-content">
+  <div class="inner-bg">
+      <div class="container">
+          <div class="row">
+              <div class="col-sm-6 col-sm-offset-3 form-box">
+                <div class="form-top">
+                  <div class="form-top-left">
+                    <h3>Login to our site</h3>
+                    <p class="unsigned-user-paragraph">Enter your username and password to log on:</p>
```
Comment (leonardk1) [inline app/views/devise/sessions/new.html.erb]: @godfreym `to log in` would be the correct phrase not `to log on`

**Comment by leonardk1** [comment]
@godfreym :sweat: 
wow, code has not issue, though still it could have been broken down
into signin, confirmation, and signout as separate PRs

**Comment by godfreym** [comment]
@arafatm Please Review

**Comment by arafatm** [comment]
@godfreym agreed with @leonardk1. Could have been broken up a bit.

Go ahead and merge

**Comment by godfreym** [comment]
:-1: @godfreym For not realising he could break this PR a bit

@arafatm @leonardk1 Will do that next time

**Pull request merged**